### DISCUSSION TITLE
Draw visualizer should have a search textbox

### DISCUSSION THREAD

**Discussion by Noordles** [ppy/osu-framework#5125] [Ideas] [answered]
Currently it's difficult and unintuitive to look for drawables in large containers (e.g. the ruleset's playfield).
Having a text box where you can filter out the name of the drawables would allow developers a quicker way to find a drawable.

**Comment by frenzibyte** [comment]
You're welcome to give it a try yourself, you can look at `SearchContainer` for a quick start. Don't hesitate to reach out on discord if you face any problems along the way.

**Reply by bdach** [reply]
Note that this might not be as simple as just hooking up search container like that. Search in a tree view is a bit more complicated than in a flat list. You'll want to figure out what to do with a node that matches the filter, but none of its parents do.

**Reply by peppy** [reply]
There's an example of how this can be done in the settings implementation osu!-side. Though whether it will scale in performance is to be seen.

I've wanted a search box for draw vis for a while now, think this thought at least once a week.

**Reply by naoei** [reply]
I've been working on this for a bit but I've come across the question of what's the expectation when an item is filtered out. My question is: should the item disappear completely or should it hide the name?

E.g.

| Disappear | Hidden |
|-----------|--------|
| <video src="https://user-images.githubusercontent.com/35349837/174502141-2fc988ec-d94d-461e-8cd1-938085a951a6.mp4"> | <video src="https://user-images.githubusercontent.com/35349837/174502178-1d479ed2-a52c-4d91-b543-d5fc6b9d4fa5.mp4"> |

**Reply by frenzibyte** [reply]
I would definitely prefer hiding filtered items completely from the hierarchy, rather than just fading them away in place, which feels more as if it's bugged.

**Comment by smoogipoo** [comment, marked as answer]
Resolved via https://github.com/ppy/osu-framework/pull/5254
